testutil/mocking_test.go

```diff
@@ -0,0 +1,62 @@
+package testutil_test
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/testutil"
+)
+
+func ExampleBackupBeforeMocking_mockingSimple() {
```

Comment:
the example are alright but they pass even if restore is wrong for example. This needs also some non-example tests checking the beavior

Reply:
Hi @pedronis,

I am not sure I understand you. These testing examples check that the restoration was correct, since they explicitly anticipate the printed values to be as in original objects. 
Can you please elaborate what you mean?

Thank you,
Arseniy